lib/src/executables/boilerplate_upgrade.dart

```diff
@@ -76,20 +99,25 @@ void main(List<String> args) {
   final shouldTreatAllComponentsAsPrivate =
       parsedArgs[_treatAllComponentsAsPrivateFlag] == true;
 
-  final query = FileQuery.dir(
-    pathFilter: (path) {
-      return isDartFile(path) && !isGeneratedDartFile(path);
-    },
-    recursive: true,
-  );
+  final path = parsedArgs['path'];
+  // If a path or glob is provided as an arg, get file paths for just that path or glob.
+  Iterable<String> fileQuery = path != null
```

Comment:
I spent >1 hour trying to test this by adding a new unit test suite that runs the `boilerplate_upgrade` executable, but ran into issues getting past the first `✔ Are these ranges correct? (Y/n)` message using `Process.start` and `stdin`.
There aren't any existing unit tests for this executable, so I decided it wasn't that important, but if you feel it's critical to test this, I can try again.